### PR TITLE
fix: adds ConfirmSignInOptions for confirmSignIn API method

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds ConfirmSignInOptions for Auth.confirmSignIn API. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthConfirmSignInOptions.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (richardmcclellan) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthConfirmSignInOptions.java:2]:
```suggestion
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: core/src/main/java/com/amplifyframework/auth/options/AuthConfirmSignInOptions.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (richardmcclellan) [inline core/src/main/java/com/amplifyframework/auth/options/AuthConfirmSignInOptions.java:2]:
```suggestion
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

**Review by richardmcclellan — COMMENTED** [review]
Looks great so far @haverchuck!   Now we just need unit tests and we should ideally add this API to the RxBindings as well, though if you wanted to do the RxBindings work as a separate PR, I'd be okay with that.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthConfirmSignInOptions.java (hunk 1 above)
Comment (haverchuck) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthConfirmSignInOptions.java:2]: Nice catch! Made this change on my end.

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java [hunk 3]
```diff
@@ -335,7 +336,7 @@ public void signIn() throws AuthException {
      */
     @Test
     @SuppressWarnings("unchecked") // Casts final parameter to Callback to differentiate methods
-    public void confirmSignIn() throws AuthException {
+    public void confirmSign() throws AuthException {
```
Comment (richardmcclellan) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java:339]:
```suggestion
    public void confirmSignIn() throws AuthException {
```

Comment (haverchuck) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java]: Fixed.

**Review by richardmcclellan — APPROVED** [review]
🚀

**Pull request merged**